src/components/Tweet/Avatar/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { createUseStyles } from 'react-jss';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { useTheme } from '../../../contexts/ThemeContext.jsx';
+import { randomColor } from '../../../util/randomColor.js';
+
+const useStyles = createUseStyles({
+  root: {
+    display: 'inline-flex',
+    textDecoration: 'none',
+  },
+  avatarCircle: {
+    width: 40,
+    height: 40,
+    backgroundColor: ({ color }) => color,
+    marginRight: 10,
+    borderRadius: 20,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    fontSize: 20,
+    color: 'white',
+    boxSizing: 'border-box',
+    '&:hover': {
+      border: ({ theme }) => `2px solid ${theme.primary.main}`,
+      cursor: 'pointer',
+    },
+  },
+  nameContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    color: ({ theme }) => theme.translucent[60],
+  },
+  displayName: {
+    fontSize: 15,
+    fontWeight: 500,
+    marginBottom: 3,
+    color: ({ theme }) => theme.translucent[90],
+    '&:hover': {
+      textDecoration: 'underline',
+    },
+  },
+  username: {
+    fontSize: 13,
+    '&:hover': {
+      textDecoration: 'underline',
+    },
+  },
+});
+
+function Avatar({ id, username, displayName, color = randomColor() }) {
```

Comment:
There is a bug on this line -- anyone have thoughts on a better approach? 🐛 

https://user-images.githubusercontent.com/15637664/173204042-5890658e-c021-48c5-ad0b-60cb690de1c5.mov

Reply:
Hmm, every time the component renders a new random color is set. Should we be passing the user preferred color as a prop instead?

Reply:
Would this work? 
What if  in the `<SignUpPopover/>` component when we create a user, which we only do once,  we also `setUser() `with a color? I think this way we would only call randomColor once() in the` handleSubmitClick() `event handler ex `const color= randomColor()`.
 We would also have to modify the current user data structure to have a color property. We could then access the color property in the `<Avatar/>` props.